### PR TITLE
feat: adding interchain account address query to controller submodule

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Adding interchain account address query to controller submodule
- Adding basic CLI client query 

closes: #2192 
needed for: #2039 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]
Aye nice work! You know what I'm going to say, needs a changelog entry :)

File: proto/ibc/applications/interchain_accounts/controller/v1/query.proto [hunk 1]
```diff
@@ -5,16 +5,34 @@ package ibc.applications.interchain_accounts.controller.v1;
...
 import "ibc/applications/interchain_accounts/controller/v1/controller.proto";
+import "gogoproto/gogo.proto";
 import "google/api/annotations.proto";
 
 // Query provides defines the gRPC querier service.
 service Query {
+  // InterchainAccount returns the interchain account address for a given owner address on a given connection
+  rpc InterchainAccount(QueryInterchainAccountRequest) returns (QueryInterchainAccountResponse) {
+    option (google.api.http).get =
+        "/ibc/apps/interchain_accounts/controller/v1/owner/{owner}/connection/{connection_id}";
```
Comment (crodriguezvega) [inline proto/ibc/applications/interchain_accounts/controller/v1/query.proto:16]: Should we use the more REST-ish style of plural nouns like we have in other routes?

```suggestion
        "/ibc/apps/interchain_accounts/controller/v1/owners/{owner}/connections/{connection_id}";
```

File: modules/apps/27-interchain-accounts/controller/client/cli/query.go [hunk 2]
```diff
@@ -11,6 +11,40 @@ import (
 	"github.com/cosmos/ibc-go/v5/modules/apps/27-interchain-accounts/controller/types"
 )
 
+// GetCmdQueryInterchainAccount returns the command handler for the controller submodule parameter querying.
+func GetCmdQueryInterchainAccount() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "interchain-account [connection-id] [owner]",
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/controller/client/cli/query.go:17]: Should we maybe swap the order?
```suggestion
		Use:     "interchain-account [owner] [connection-id]",
```

**Review by crodriguezvega — COMMENTED** [review]
Thank you, @damiannolan!

I left a couple of suggestions.

~~Should we add some documentation about this new query in the docs?~~ Ah, never mind, I saw we have [this other issue](https://github.com/cosmos/ibc-go/issues/1948).

File: modules/apps/27-interchain-accounts/controller/keeper/grpc_query.go [hunk 3]
```diff
@@ -4,16 +4,41 @@ import (
...
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	portID, err := icatypes.NewControllerPortID(req.Owner)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to generate portID from owner address: %s", err)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/controller/keeper/grpc_query.go:27]: yeah I mostly agree with you tbh, but its kind of inconsistent across a lot of our grpc queries, maybe we can align them all in another PR.

**Comment by colin-axner** [comment]
Can we backport this to minor releases? Would allow anyone launching pre v6 to start using this gRPC and avoid migrations later cc @crodriguezvega

**Review by chatton — APPROVED** [review]
LGTM nice work 🥇

File: modules/apps/27-interchain-accounts/controller/client/cli/query.go [hunk 4]
```diff
@@ -11,6 +11,40 @@ import (
...
 )
 
+// GetCmdQueryInterchainAccount returns the command handler for the controller submodule parameter querying.
+func GetCmdQueryInterchainAccount() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "interchain-account [owner] [connection-id]",
+		Short:   "Query the interchain account address for a given owner on a particular connection",
+		Long:    "Query the controller submodule for the interchain account address for a given owner on a particular connection",
+		Args:    cobra.ExactArgs(2),
+		Example: fmt.Sprintf("%s query interchain-accounts controller interchain-account connection-0 cosmos1layxcsmyye0dc0har9sdfzwckaz8sjwlfsj8zs", version.AppName),
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/client/cli/query.go:21]: fix ordering in example string: 

```suggestion
		Example: fmt.Sprintf("%s query interchain-accounts controller interchain-account cosmos1layxcsmyye0dc0har9sdfzwckaz8sjwlfsj8zs connection-0", version.AppName),
```

**Review by AdityaSripal — APPROVED** [review]

**Pull request merged**